### ISSUE TITLE
[BUG] Resume from checkpoint Out of memory error (SIGTERM: Killed) for a large model

### ISSUE THREAD

**Issue by ozanciga** [microsoft/DeepSpeed#3104]
**Describe the bug**

- Using the latest version of transformers and deepspeed.
- It is possible to fit a ~1.5B parameter model and train it from scratch with batch sizes up to 32 (with gradient checkpointing, bf32 type) on a single 3090 RTX, which is amazing btw.
- However, if I try to resume from a checkpoint with even a batch size of 1, there is an OOM error. The message is SIGTERM: Killed.
- I believe this is a bug, since the model can easily fit (the training takes up only 22Gs!) if you start a fresh job, but doesn't work with continuing a job. Also, it is really painful having to restart a job that you trained for 2 weeks, knowing that (verified by performing inference on the checkpoint with good results) your weights are valid, but useless for further training.

**To Reproduce**
I'm working with a custom diffusion model, however you can verify this with following models of similar size: OPT-1.3b, whisper-large, Stable Diffusion 1.4. Train for 10 iterations, save a checkpoint. Then restart. The memory consumption jumps to the maximum, then OOM.

**Expected behavior**

**ds_report output**
Please run `ds_report` to give us details about your setup.

**System info (please complete the following information):**
 - OS: Ubuntu 22
 - GPU count and types: Single RTX 3090 (24G)
 - Python version: 3.9
 - 64G RAM, AMD Ryzen 3950x

**Launcher context**
deepspeed train.py

Thank you!

**Comment by ozanciga** [comment]
to add to this, every time i save a checkpoint with optimizer states, i notice an increase in ram use. at a certain point, oom and the script crashes. as above. i see this as a bug since it is possible to start a fresh training job with the same weights without crashing but throughout the training the memory use increases despite batch size staying the same.

**Comment by tjruwase** [comment]
@ozanciga, can you please share a stack trace of the OOM? Is the OOM CPU or GPU? Can you share run script? 

We may not be able to repro the OOM since our hardware is different, but the above information can help the investigation.

**Comment by RameshArvind** [comment]
You can try following the [suggestion](https://github.com/huggingface/transformers/issues/17258#issuecomment-1128905263) of adding more memory via swap file. We faced a similar issue before and that resolved it

**Comment by tjruwase** [comment]
@ozanciga, sorry that you are losing your training cycles due to inability to load the checkpoints. 

I think the problem might be related to https://github.com/microsoft/DeepSpeed/issues/3303. One way to confirm is to inspect the checkpoint folder size. Can you please share the results of running `du -a -h --max-depth=1` on your checkpoint folder?
